Description: make install name of applications.menu file a cached cmake variable
 To not conflict with kde4runtime nor gnome nor xfce, distributions may
 choose to have a different name for their applications.menu file.
 To enable this without pitchy patching make the installed name fully
 parameterized via the cmake variable APPLICATIONS_MENU_NAME.
Author: Harald Sitter <sitter@kde.org>
Origin: vendor
Forwarded: https://git.reviewboard.kde.org/r/125284/
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: kservice/CMakeLists.txt
===================================================================
--- kservice.orig/CMakeLists.txt
+++ kservice/CMakeLists.txt
@@ -28,6 +28,10 @@ ecm_setup_version(${KF5_VERSION}
                   PACKAGE_VERSION_FILE "${CMAKE_CURRENT_BINARY_DIR}/KF5ServiceConfigVersion.cmake"
                   SOVERSION 5)
 
+set(APPLICATIONS_MENU_NAME applications.menu
+    CACHE STRING "Name to install the applications.menu file as.")
+add_definitions(-DAPPLICATIONS_MENU_NAME="${APPLICATIONS_MENU_NAME}")
+
 # TODO: Remove these
 remove_definitions(-DQT_NO_CAST_FROM_ASCII)
 
Index: kservice/src/CMakeLists.txt
===================================================================
--- kservice.orig/src/CMakeLists.txt
+++ kservice/src/CMakeLists.txt
@@ -135,9 +135,9 @@ install(FILES
 )
 
 if (WIN32)
-install( FILES applications.menu  DESTINATION  ${KDE_INSTALL_DATAROOTDIR}/xdg/menus )
+install( FILES applications.menu  DESTINATION  ${KDE_INSTALL_DATAROOTDIR}/xdg/menus  RENAME ${APPLICATIONS_MENU_NAME} )
 else ()
-install( FILES applications.menu  DESTINATION  ${KDE_INSTALL_SYSCONFDIR}/xdg/menus )
+install( FILES applications.menu  DESTINATION  ${KDE_INSTALL_SYSCONFDIR}/xdg/menus  RENAME ${APPLICATIONS_MENU_NAME} )
 endif ()
 
 install(TARGETS KF5Service EXPORT KF5ServiceTargets ${KF5_INSTALL_TARGETS_DEFAULT_ARGS})
Index: kservice/src/sycoca/kbuildsycoca.cpp
===================================================================
--- kservice.orig/src/sycoca/kbuildsycoca.cpp
+++ kservice/src/sycoca/kbuildsycoca.cpp
@@ -274,7 +274,7 @@ bool KBuildSycoca::build()
             m_vfolder->setTrackId(m_trackId);
         }
 
-        VFolderMenu::SubMenu *kdeMenu = m_vfolder->parseMenu(QStringLiteral("applications.menu"));
+        VFolderMenu::SubMenu *kdeMenu = m_vfolder->parseMenu(QStringLiteral(APPLICATIONS_MENU_NAME));
 
         KServiceGroup::Ptr entry = m_buildServiceGroupFactory->addNew(QStringLiteral("/"), kdeMenu->directoryFile, KServiceGroup::Ptr(), false);
         entry->setLayoutInfo(kdeMenu->layoutList);
